pkg/analysis/optionalfields/util.go

```diff
@@ -143,3 +144,51 @@ func reportShouldAddOmitEmpty(pass *analysis.Pass, field *ast.Field, omitEmptyPo
 		panic(fmt.Sprintf("unknown omit empty policy: %s", omitEmptyPolicy))
 	}
 }
+
+// reportShouldAddOmitZero adds an analysis diagnostic that explains that an omitzero tag should be added.
+func reportShouldAddOmitZero(pass *analysis.Pass, field *ast.Field, omitZeroPolicy OptionalFieldsOmitZeroPolicy, fieldName, messageFmt string, fieldTagInfo extractjsontags.FieldTagInfo) {
+	switch omitZeroPolicy {
+	case OptionalFieldsOmitZeroPolicySuggestFix:
+		pass.Report(analysis.Diagnostic{
+			Pos:     field.Pos(),
+			Message: fmt.Sprintf(messageFmt, fieldName),
+			SuggestedFixes: []analysis.SuggestedFix{
+				{
+					Message: fmt.Sprintf("should add 'omitzero' to the field tag for field %s", fieldName),
+					TextEdits: []analysis.TextEdit{
+						{
+							Pos:     fieldTagInfo.Pos + token.Pos(len(fieldTagInfo.Name)),
+							NewText: []byte(",omitzero"),
+						},
+					},
+				},
+			},
+		})
+	case OptionalFieldsOmitZeroPolicyWarn:
+		pass.Reportf(field.Pos(), messageFmt, fieldName)
+	case OptionalFieldsOmitZeroPolicyIgnore, OptionalFieldsOmitZeroPolicyForbid:
+		// Do nothing, as the policy is to ignore the missing omitzero tag.
+	default:
+		panic(fmt.Sprintf("unknown omit zero policy: %s", omitZeroPolicy))
+	}
+}
+
+// reportShouldRemoveOmitZero adds an analysis diagnostic that explains that an omitzero tag should be removed.
+func reportShouldRemoveOmitZero(pass *analysis.Pass, field *ast.Field, fieldName string, jsonTags extractjsontags.FieldTagInfo) {
+	pass.Report(analysis.Diagnostic{
+		Pos:     field.Pos(),
+		Message: fmt.Sprintf("field %s has omitzero tag but forbid omitzero policy is configured for linter so omitzero tag should be removed", fieldName),
+		SuggestedFixes: []analysis.SuggestedFix{
+			{
+				Message: "should remove the omitzero tag",
+				TextEdits: []analysis.TextEdit{
+					{
+						Pos:     jsonTags.Pos,
+						End:     jsonTags.End,
+						NewText: []byte(strings.Replace(jsonTags.RawValue, ",omitzero", "", 1)),
+					},
```

Comment:
Either something wrong with this replacement or we can't do both replace and add to tag at the same time, 
Test is failing for a field in b/a.go with Omitzero forbid policy. Any inputs here?

Field
```
StructWithOnlyOmitZero *D `json:"structWithOnlyOmitZero,omitzero"`
```

Expected
```
StructWithOnlyOmitZero *D `json:"structWithOnlyOmitZero,omitempty"`
```

Actual
```
 StructWithOnlyOmitZero *D `json:",omitemptystructWithOnlyOmitZero"`
```

Reply:
It's possible that we might have to combine this case into one suggestion, I've not sen us use overlapping fixes before. It could also be an ordering issue, have you tried running the tests many times to see if it passes sometimes but not on others?

Reply:
I had a play with this locally and worked out a solution.

Add the variable at the top of the function

```
omitZeroPos := jsonTags.Pos + token.Pos(strings.Index(jsonTags.RawValue, ",omitzero"))
```

Then the fix should be 

```
{
	Pos:     omitZeroPos,
	End:     omitZeroPos + token.Pos(len(",omitzero")),
	NewText: nil, // Clear the omitzero tag.
},
```